### PR TITLE
Fix NullPointerException in VideoPlayer

### LINKED ISSUE #5911: [ACRA 2.9] - NullPointerException: VideoPlayer/SoundPlayer
https://couchdb.ankidroid.org/acralyzer/_design/acralyzer/index.html#/report-details/311a66d3-8c15-4c1a-b87d-6a31e678a083

```
java.lang.NullPointerException: uriString
at android.net.Uri$StringUri.<init>(Uri.java:475)
at android.net.Uri$StringUri.<init>(Uri.java)
at android.net.Uri.parse(Uri.java:437)
at com.ichi2.libanki.Sound.playSound(Sound.java:3)
at com.ichi2.anki.VideoPlayer.surfaceCreated(VideoPlayer.java:1)
at android.view.SurfaceView.updateWindow(SurfaceView.java:586)
at android.view.SurfaceView$3.onPreDraw(SurfaceView.java:177)
at android.view.ViewTreeObserver.dispatchOnPreDraw(ViewTreeObserver.java:944)
at android.view.ViewRootImpl.performTraversals(ViewRootImpl.java:2055)
at android.view.ViewRootImpl.doTraversal(ViewRootImpl.java:1107)
at android.view.ViewRootImpl$TraversalRunnable.run(ViewRootImpl.java:6013)
at android.view.Choreographer$CallbackRecord.run(Choreographer.java:858)
at android.view.Choreographer.doCallbacks(Choreographer.java:670)
at android.view.Choreographer.doFrame(Choreographer.java:606)
at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:844)
at android.os.Handler.handleCallback(Handler.java:746)
at android.os.Handler.dispatchMessage(Handler.java:95)
at android.os.Looper.loop(Looper.java:148)
at android.app.ActivityThread.main(ActivityThread.java:5421)
at java.lang.reflect.Method.invoke(Native Method)
at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
```

### PR DESCRIPTION
**Pull request by david-allison-1**
## Purpose / Description

#5911 raised an unexpected error in VideoPlayer, we log to see how this happened, and attempt to fix it

## Fixes
Fixes #5911 

## Approach
Null check & Logging

## How Has This Been Tested?

Couldn't reproduce or logically reason through how it happened. 🛑 

## Checklist
- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [x] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code

### CODE REVIEW

File: AnkiDroid/src/main/java/com/ichi2/anki/VideoPlayer.java [hunk 1]
```diff
@@ -35,26 +36,36 @@
...
-                finish();
-                MediaPlayer.OnCompletionListener originalListener = mSoundPlayer.getMediaCompletionListener();
-                if (originalListener != null) {
-                    originalListener.onCompletion(mp);
-                }
+        Timber.i("surfaceCreated");
+
+        if (mPath == null) {
+            //#5911 - path shouldn't be null. I couldn't determine why this happens.
+            //TODO: We should catch this via a diagnostic event
```
Comment (david-allison-1) [inline AnkiDroid/src/main/java/com/ichi2/anki/VideoPlayer.java:57]: Flagging for review, wasn't sure how to do this via ACRA as we don't have an exception.

**Comment by timrae** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- AnkiDroid/src/main/java/com/ichi2/anki/VideoPlayer.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5298557&bid=17343405)

File: AnkiDroid/src/main/java/com/ichi2/anki/VideoPlayer.java (hunk 1 above)
Comment (mikehardy) [reply AnkiDroid/src/main/java/com/ichi2/anki/VideoPlayer.java:57]: You can sendExceptionReport with a new Exception instantiated right there

Comment (david-allison-1) [reply AnkiDroid/src/main/java/com/ichi2/anki/VideoPlayer.java:57]: I'll add a helper function. Let's flag as awaiting feedback

Comment (mikehardy) [reply AnkiDroid/src/main/java/com/ichi2/anki/VideoPlayer.java:57]: sendExceptionReport is already a helper function :-), just do a `new Exception("my cool message")` right inside of it and you're set

Comment (david-allison-1) [reply AnkiDroid/src/main/java/com/ichi2/anki/VideoPlayer.java]: Done

**Comment by david-allison-1** [comment]
Requesting re-review (mostly on the design of the exception reporting API).

File: AnkiDroid/src/test/java/com/ichi2/anki/AnkiDroidAppTest.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+@RunWith(AndroidJUnit4.class)
+public class AnkiDroidAppTest {
+
+    @Test
+    public void reportingDoesNotThrowException() {
+        AnkiAssert.assertDoesNotThrow(() -> sendExceptionReport("Test", "AnkiDroidAppTest"));
+    }
+
+    @Test
+    public void reportingWithNullMessageDoesNotFail() {
```
Comment (timrae) [inline AnkiDroid/src/test/java/com/ichi2/anki/AnkiDroidAppTest.java:22]: Codacy Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5298557)

**Comment by timrae** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- AnkiDroid/src/test/java/com/ichi2/anki/AnkiDroidAppTest.java  1
- AnkiDroid/src/main/java/com/ichi2/anki/exception/ManuallyReportedException.java  1
- AnkiDroid/src/main/java/com/ichi2/anki/VideoPlayer.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5298557&bid=17343405)

File: AnkiDroid/src/test/java/com/ichi2/anki/AnkiDroidAppTest.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+import androidx.test.ext.junit.runners.AndroidJUnit4;
+
+import static com.ichi2.anki.AnkiDroidApp.sendExceptionReport;
+
+@RunWith(AndroidJUnit4.class)
+public class AnkiDroidAppTest {
+
+    @Test
+    public void reportingDoesNotThrowException() {
```
Comment (timrae) [inline AnkiDroid/src/test/java/com/ichi2/anki/AnkiDroidAppTest.java:17]: Codacy Issue found: [JUnit tests should include assert() or fail()](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5298557)

**Pull request merged**